Added SLES12 SP5 and SLES4SAP

deankannenberg opened the pull request:
## What does this PR change?

Add the needed onboarding repos entires for SLES 12 SP5 and SLES4 SAP products to spacewalk-common-channels.

- add SLES12 SP5 support for bootstrap repos
- add SLES4SAP (12 SP4, 12 SP5, 15SP2) support for bootstrap repos

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: We don't document all channels at spacewalk-common-channels.

- [x] **DONE**

## Test coverage
- No tests: Manually tested

- [x] **DONE**

## Links

None

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)


## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

juliogonzalez reviewed (changes requested):
@deankannenberg I updated the template at the description for you. I guessed you tested this manually, so that's what I wrote.

The changes look good. I guess you are not adding SLES4SAP 15SP1, 12SP4 etc because the are on ESPOS, right? (https://www.suse.com/lifecycle/)

The only missing thing is the changelog entry.

You just need to edit `utils/spacewalk-utils` and add a new entry on top.

Something like, for example:

```
- Add client tools for SLES12SP5 and SLES4SAP 12SP4/12SP5/15SP2
```

deankannenberg commented:
Thank you for helping with my first contribution.
Yes I only added the SLES 12 SP4 / SLES 12 SP5 / SLES 15 SP2 client tools because of the current SLES lifecycle.
I updated the "utils/spacewalk-utils.changes" file. I hope I changed it the right way.

juliogonzalez commented on utils/spacewalk-common-channels.ini:
```diff
@@ -1221,6 +1281,26 @@ gpgkey_id = %(_uyuni_gpgkey_id)s
+archs    =  x86_64, s390x, aarch64, ppc64le
+base_channels = sle-product-sles_sap15-sp2-pool-%(arch)s
+checksum = sha256
+gpgkey_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Master:/SLE15-Uyuni-Client-Tools/SLE_15/repodata/repomd.xml.key
```
I guess the URL should be `https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/SLE15-Uyuni-Client-Tools/SLE_15/repodata/repomd.xml.key` here (Stable tools)...

deankannenberg replied:
I copied this url block from the other SLES15 SP2 parts above. Shall I just edit this from Master to Stable anyway?
See for e.g. line: 1249

juliogonzalez replied:
Seems you unearthed a bug. Yes, please, change this line to Stable, and same for 1292. I will create a PR to fix SLE15SP2 above.

deankannenberg replied:
changed from master to stable in dd89b5f

juliogonzalez replied:
The PR to fix the other repositories is at https://github.com/uyuni-project/uyuni/pull/3636

juliogonzalez commented on utils/spacewalk-common-channels.ini:
```diff
@@ -1221,6 +1281,26 @@ gpgkey_id = %(_uyuni_gpgkey_id)s
+gpgkey_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Master:/SLE15-Uyuni-Client-Tools/SLE_15/repodata/repomd.xml.key
+gpgkey_id = %(_uyuni_gpgkey_id)s
+gpgkey_fingerprint = %(_uyuni_gpgkey_fingerprint)s
+repo_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Master:/SLE15-Uyuni-Client-Tools/SLE_15/
```
And `https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/SLE15-Uyuni-Client-Tools/SLE_15/` here.

deankannenberg replied:
I copied this url block from the other SLES15 SP2 parts above. Shall I just edit this from Master to Stable anyway?
See for e.g. line: 1252

deankannenberg replied:
changed from master to stable in dd89b5f

juliogonzalez reviewed (commented):
Changelog is fine :-)

I just noticed something now about SLE15 SAP SP2. Other than that, the PR LGTM.

Thanks for the contribution!

@mcalmer, can you have a second look tomorrow, just in case?

juliogonzalez reviewed (approved):
Looks good!

Thanks again for the contribution.

mcalmer reviewed (approved):
Looks good so far.
Maybe something for a future PR: add this also for SLES15-SP3* ?

deankannenberg commented:
I just added the SLES4SAP 15 SP3 client tools too.
See: 
f17d6a30c9763e2bfd5987d947964a8f0a81a8ec
ee7e4c6cd6280570e3bcc57d84365f627530cbe0

juliogonzalez commented on utils/spacewalk-common-channels.ini:
```diff
@@ -1241,6 +1321,26 @@ gpgkey_id = %(_uyuni_gpgkey_id)s
+archs    =  x86_64, s390x, aarch64, ppc64le
+base_channels = sle-product-sles_sap15-sp3-pool-%(arch)s
+checksum = sha256
+gpgkey_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/SLE15-Uyuni-Client-Tools/SLE_15/repodata/repomd.xml.key
```
```suggestion
gpgkey_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Master:/SLE15-Uyuni-Client-Tools/SLE_15/repodata/repomd.xml.key
```

(as this is for the devel client tools)

juliogonzalez replied:
```suggestion
gpgkey_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Master:/SLE15-Uyuni-Client-Tools/SLE_15/repodata/repomd.xml.key
```

(as this is for the devel client tools)

juliogonzalez commented on utils/spacewalk-common-channels.ini:
```diff
@@ -1241,6 +1321,26 @@ gpgkey_id = %(_uyuni_gpgkey_id)s
+gpgkey_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/SLE15-Uyuni-Client-Tools/SLE_15/repodata/repomd.xml.key
+gpgkey_id = %(_uyuni_gpgkey_id)s
+gpgkey_fingerprint = %(_uyuni_gpgkey_fingerprint)s
+repo_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/SLE15-Uyuni-Client-Tools/SLE_15/
```
```suggestion
repo_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Master:/SLE15-Uyuni-Client-Tools/SLE_15/
```

juliogonzalez reviewed (changes requested):
Two more changes, and ready to merge :-)

juliogonzalez reviewed (approved):
Merging as soon as tests are passing.

juliogonzalez commented:
... and merged!

Thanks a lot @deankannenberg :-)

This will be part of 2021.05
